### DISCUSSION TITLE
wireguard hard drive activity

### DISCUSSION THREAD

**Discussion by ppeterrr** [qdm12/gluetun#751] [Q&A]
I'm running wireguard along with the http proxy on a synology 920 in a container. Excellent, thanks for that!

It only strikes me that when I start gluetun, there is continuously 300 to 400kbs activity on the hdds. Well, that's probably normal, but I'd like to minimize that so I don't hear the clatter of the discs all day long.

I've disabled the log options, which didn't really help. I'm thinking of running docker on an nvme drive, but maybe there's some other advice?

Thanks

**Comment by qdm12** [comment]
Hi there! That's really strange, it's not meant to write anything to disk (well except to stdout) constantly. It even rarely writes anything to disk. Maybe it's the Alpine system that does it 🤔 Is it using the userspace or kernelspace wireguard implementation (check in your logs)?

**Reply by ppeterrr** [reply]
Hey, thanks for your swift response. I see this in the logs,  i'm kind of new to this, so it doesnt tell me much.

Does it have to do with some missing binaries in the kernel ?

`wireguard: Using userspace implementation since Kernel support does not exist`

[edit] I might need to compile that runfalk package then, not to comfy about that. But i think i can manage :)

**Reply by ppeterrr** [reply]
So, it took me a while but i installed the binaries to run wireguard with kernel support in DSM 7

`2021/12/17 14:57:43 INFO wireguard: Using available kernelspace implementation`

But still. when i start gluetun, i get around 300kbs read/write activity. When i stop the container it directly stops.

**Reply by qdm12** [reply]
Yeah it's probably unrelated to the Wireguard implementation used, my bad, userspace and kernelspace shouldn't write anything AFAIK.

**Comment by ppeterrr** [comment]
@qdm12 

I hve been trying to solve this problem, and i didnt manage yet :)

But i can tell you, that i think it is the docker service from the synology thats giving hdd activity when i start the container.
So it is probably not gleutun, but maybe the way i configged it....not sure yet

![Schermafbeelding 2021-12-19 151352](https://user-images.githubusercontent.com/96293249/146680920-c233e21c-1677-4248-a70c-8f84ef89d7b3.jpg)

**Comment by ppeterrr** [comment]
Oke guys.  I've not found a reason or a solution. But the IO rumble on the drives
irritated me a lot. So, i made a dirty workaround.

Migrated the docker package and the container to a 2Tb Nvme. Noise is gone, IO is still there.

Hopefully one day i will find out what i did wrong. Or, what is going on with Gluetun giving me extra IO.

Thanks for now.

**Reply by qdm12** [reply]
Yeah I put all my programs on SSDs as well to avoid noise :smile: Sorry for the late reply;

Did you have disk activity with gluetun not running? Since it looks like it's coming from `dockerd`?
If you have disk activity only with Gluetun, you can try disabling the healthcheck perhaps see if it makes a difference?

**Reply by ppeterrr** [reply]
> Yeah I put all my programs on SSDs as well to avoid noise 😄 Sorry for the late reply;
> 
> Did you have disk activity with gluetun not running? Since it looks like it's coming from `dockerd`? If you have disk activity only with Gluetun, you can try disabling the healthcheck perhaps see if it makes a difference?

I am almost sure it is the docker deamon. Docker on the synology is not really vanilla. And maybe
there is some extra logging from the docker. package  Not sure. There is no other IO when gluetun is not running.
And other containers dont give the IO rumble on the drives.

I tried disabling everything, also healthcheck. Didnt make a difference.

For now its good enough for me, glad that i now can leave the proxy running and the wireshark connection up
without the irritating noise :)

If there comes up a reason, or maybe you find out later on what gives the extra IO, i'll be glad to test it out for you ofcourse.

Thanks, cheers :)

**Reply by qdm12** [reply]
So I (finally) had a look at `docker stats`. Mine doesn't really show hard disk activity, I get the legend:

```
CONTAINER ID   NAME                        CPU %     MEM USAGE / LIMIT     MEM %     NET I/O          BLOCK I/O         PIDS
```

And block i/o doesn't really move. So it's hard to tell.

On the other hand I noticed memory usage goes to 300 / 500 MB which is ridiculous, I definitely need to investigate how to reduce this! Especially when it uses the Wireguard kernel implementation, like come on it's not really doing anything :smile: 
It might be due to the built-in server data being loaded in memory perhaps, I'll check. EDIT: created #765
